### PR TITLE
fix(devspace): explicitly set requests/limits on devspace container

### PR DESCRIPTION
**Pull request by malept**
## What this PR does / why we need it

This should fix problems some people were having with OOM kills in various places by default.

## Jira ID

[DT-3950]

## Notes for your reviewers

Ignore the whitespace changes, those were autogenerated by VSCode.

[DT-3950]: https://outreach-io.atlassian.net/browse/DT-3950?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/devspace.yaml.tpl [hunk 1]
```diff
@@ -94,6 +94,13 @@ vars:
 deployments:
   app:
     namespace: ${DEVENV_DEPLOY_NAMESPACE}
+    resources:
+      requests:
+        cpu: 1
+        memory: 1Gi
```
Comment (joeysmithoutreach) [inline templates/devspace.yaml.tpl:100]: Do we know how this behaves in a regular devenv environment when multiple services use this as their default? The change looks good, but my only worry is that by setting this as a minimum we might limit how many small services people could run at one time.

Comment (malept) [reply templates/devspace.yaml.tpl:100]: I think the question becomes, how many devspace containers do we expect folks to have running at once?

**Review by joeysmithoutreach — APPROVED** [review]

File: templates/devspace.yaml.tpl (hunk 1 above)
Comment (malept) [reply templates/devspace.yaml.tpl:100]: We talked about it offline - to be clear this only affects deploying apps into the devenv from source (e.g., `devenv apps run`).

File: templates/devspace.yaml.tpl [hunk 2]
```diff
@@ -94,6 +94,13 @@ vars:
...
+      limits:
+        cpu: 5
+        memory: 10Gi
```
Comment (malept) [inline templates/devspace.yaml.tpl:103]: This was a pretty arbitrary set of values, we can change it if it causes problems.

**Pull request merged**